src/scripts/blocks/countdown-timer/DisplayComponent.js

```diff
@@ -0,0 +1,177 @@
+import classnames from 'classnames';
+
+const randId = () => Math.random().toString(36).replace(/[^a-z]+/g, '').substr(2, 10);
+
+const { __ } = wp.i18n;
+const { Component, Fragment } = wp.element;
+const {
+  InspectorControls, RichText, URLInputButton,
+} = wp.editor;
+const {
+  PanelBody, Button, DateTimePicker, ColorPicker, SelectControl, ToggleControl,
+} = wp.components;
+const { applyFilters } = wp.hooks;
+const { PostMediaSelector } = benenson.components;
+
+class DisplayComponent extends Component {
+  componentDidMount() {
+    const { attributes, setAttributes } = this.props;
+
+    if (!attributes.countdownId) {
+      setAttributes({
+        countdownId: randId(),
+      });
+    }
+  }
+
+  render() {
+    const { attributes, setAttributes } = this.props;
+    const {
+      countdownId = '',
+      date = new Date(),
+      title = '',
+      revealTitle = '',
+      revealContent = '',
+      revealBtnText = '',
+      revealBtnUrl = '',
+      backgroundColor = '',
+      backgroundId = null,
+      backgroundUrl = '',
+      textColor = '',
+      hideTimer = false,
+    } = attributes;
+
+    // Note: US English spelling.
+    const colourOptions = applyFilters('benenson.block.blockCountdown.colorOptions', [{
+      label: __('White', 'benenson'),
+      value: '',
+    }, {
+      label: __('Black', 'benenson'),
+      value: 'black',
+    }]);
+
+    const styles = {
+      backgroundImage: `url('${backgroundUrl}')`,
+      backgroundColor,
+    };
+
+    const classes = classnames('countdownTimer', {
+      [`is-${textColor}`]: !!textColor,
+    });
+
+    return (
+      <Fragment>
+        <InspectorControls>
+          <PanelBody title={ __('Options', 'benenson') }>
+            <ToggleControl
+              label={ __('Hide timer', 'benenson') }
+              help={ __('When the timer has finished hide it from users.', 'benenson') }
```

Comment:
This would read better as `Hide the timer from users when the countdown has completed.`.